### PR TITLE
[3810] Preview Search Panel

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/3810

### CODE REVIEW

File: ui/app/src/components/Icons/LevelDescriptor.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export default createSvgIcon(
+  <>
+    <g clipPath="url(#clip0)">
+      <path d="M8.74342 4.4647C8.74342 3.9947 9.31342 3.7547 9.64842 4.0897L11.8884 6.3297C12.0834 6.5247 12.0834 6.8397 11.8884 7.0347L9.64842 9.2747C9.31342 9.6147 8.74342 9.3747 8.74342 8.9047C8.74342 8.7647 8.79842 8.6297 8.89842 8.5297L10.7434 6.6847L8.89842 4.8397C8.79842 4.7397 8.74342 4.6047 8.74342 4.4647ZM5.74342 5.1847V4.3897C5.74342 3.9447 6.28342 3.7197 6.59842 4.0347L8.89342 6.3297C9.08842 6.5247 9.08842 6.8397 8.89342 7.0347L6.59842 9.3297C6.28342 9.6447 5.74342 9.4197 5.74342 8.9747V8.1347C3.24342 8.1347 1.49342 8.9347 0.243418 10.6847C0.743418 8.1847 2.24342 5.6847 5.74342 5.1847Z" />
+    </g>
+    <defs>
+      <clipPath id="clip0">
+        <rect width="12" height="12" fill="white" transform="translate(0.0344238 0.68457)" />
+      </clipPath>
+    </defs>
```
Comment (rart) [inline ui/app/src/components/Icons/LevelDescriptor.tsx]: Can we get rid of everything but the paths?

File: ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx [hunk 2]
```diff
@@ -133,7 +130,8 @@ function SearchResults(props) {
 const initialSearchParameters: Partial<ElasticParams> = {
   keywords: '',
   offset: 0,
-  limit: 10
+  limit: 10,
+  orOperator: true
   // sortBy: '_score',
   // sortOrder: 'desc',
   // filters: {}
```
Comment (rart) [inline ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx]: ?

File: ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx [hunk 3]
```diff
@@ -144,69 +142,81 @@ const mimeTypes = ['image/png', 'image/jpeg', 'image/gif', 'video/mp4', 'image/s
...
-      //   type: 'Component'
-      // }).subscribe(
-      //   (result) => {
-      //     setSearchResults(result);
-      //   },
-      //   ({ response }) => {
-      //     setError(response);
-      //   }
-      // )
+      // pipe(takeUntil(unMount$),switchMap())
```
Comment (rart) [inline ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx]: ?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/Icons/LevelDescriptor.tsx (hunk 1 above)
Comment (aachavarria) [reply ui/app/src/components/Icons/LevelDescriptor.tsx]: yes

File: ui/app/src/components/Dialogs/BulkUploadDialog.tsx [hunk 4]
```diff
@@ -629,13 +629,6 @@ export default function BulkUploadDialog(props: BulkUploadProps) {
     e.preventDefault();
   };
 
-  const onFileUploaded = useCallback(
-    (path: string) => {
-      dispatch(emitSystemEvent(itemCreated({ target: path })));
-    },
-    [dispatch]
-  );
```
Comment (rart) [inline ui/app/src/components/Dialogs/BulkUploadDialog.tsx]: This isn't much help and takes away from the presentational nature of the dialog UI. Let's rollback.

File: ui/app/src/components/SystemStatus/MinimizedBar.tsx [hunk 5]
```diff
@@ -40,8 +40,9 @@ interface MinimizedBarProps {
 export function MinimizedBar(props: MinimizedBarProps) {
   const { title, onMaximized, subtitle, status } = props;
   const classes = useStyles({});
+
   return (
-    <Paper className={classes.root}>
+    <Paper className={classes.root} id="testeo">
```
Comment (rart) [inline ui/app/src/components/SystemStatus/MinimizedBar.tsx]: x

File: ui/app/src/components/ToolsPanelEmbeddedAppViewButton/ToolsPanelEmbeddedAppViewButton.tsx [hunk 6]
```diff
@@ -56,14 +60,48 @@ const useStyles = makeStyles((theme) =>
 export default function ToolsPanelEmbeddedAppViewButton(props) {
   const [open, setOpen] = useState(false);
   const openEmbeddedApp = () => setOpen(true);
```
Comment (rart) [inline ui/app/src/components/ToolsPanelEmbeddedAppViewButton/ToolsPanelEmbeddedAppViewButton.tsx]: Please check what happens when you...
1. open dialog
2. minimize it
3. click again on the tool panel item to open

File: ui/app/src/modules/Preview/Tools/ToolPanel.tsx [hunk 7]
```diff
@@ -63,6 +82,16 @@ export const PanelHeader: FunctionComponent<PanelHeaderProps> = (props) => {
...
+        {onMinimize && (
+          <IconButton onClick={onMinimize} className={classes.alignRight}>
+            <MinimizeIcon />
+          </IconButton>
+        )}
+        {onClose && (
+          <IconButton onClick={onClose}>
+            <CloseIcon />
+          </IconButton>
+        )}
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/ToolPanel.tsx]: This file was meant to be rolled back

File: ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx [hunk 8]
```diff
@@ -119,7 +126,8 @@ export const ComponentsPanelUI: React.FC<ComponentsPanelUIProps> = (props) => {
         pushToolsPanelPage(
           createToolsPanelPage('previewBrowseComponentsPanel.title', [
             {
-              id: 'craftercms.components.PreviewBrowseComponentsPanel'
+              id: 'craftercms.components.PreviewBrowseComponentsPanel',
+              uiKey: `uiKey_${Date.now()}`
```
Comment (rart) [inline ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx]: Add the uiKey on the reducer when not present

File: ui/guest/src/store/reducers/root.ts [hunk 9]
```diff
@@ -732,7 +722,8 @@ const reducerFunctions: {
   [HOST_CHECK_IN]: (state, action) => ({
     ...state,
     hostCheckedIn: true,
-    status: action.payload.editMode ? EditingStatus.LISTENING : EditingStatus.OFF
+    editMode: action.payload.editMode,
+    status: EditingStatus.LISTENING
```
Comment (rart) [inline ui/guest/src/store/reducers/root.ts]: No need to change the status on HostCheckIn

File: ui/app/src/services/configuration.ts [hunk 10]
```diff
@@ -196,19 +197,20 @@ export function setActiveTargetingModel(data): Observable<ActiveTargetingModel>
...
-export function getSiteUiConfig(site: string): Observable<any> {
+export function getSiteUiConfig(
+  site: string
+): Observable<Omit<GlobalState['uiConfig'], 'error' | 'isFetching' | 'currentSite'>> {
   return getConfigurationDOM(site, '/ui.xml', 'studio').pipe(
     map((xml) => {
       if (xml) {
         const widgets = xml.querySelector('[id="craftercms.components.ToolsPanel"] > configuration > widgets');
         if (widgets) {
+          widgets.querySelectorAll('widget').forEach((e, index) => e.setAttribute('uiKey', String(index)));
```
Comment (rart) [inline ui/app/src/services/configuration.ts:211]: Please add comment explaining why we had to add this:

When rendering widgets dynamically and changing pages on the tools panel, if there are duplicate react `key` props across pages, react may no swap the components correctly, incurring in unexpected behaviours. We need a unique key for each widget.

Add comment to the createwidgetdescriptor function too.

File: ui/app/src/modules/Preview/Tools/ToolPanel.tsx [hunk 11]
```diff
@@ -28,6 +28,8 @@ type ToolPanelProps = PropsWithChildren<{
   title: string | MessageDescriptor;
   BackIcon?: ElementType;
   onBack?: () => void;
+  onMinimize?: () => void;
+  onClose?: () => void;
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/ToolPanel.tsx]: x

File: ui/app/src/modules/Preview/Tools/ToolPanel.tsx [hunk 12]
```diff
@@ -47,6 +49,9 @@ const useStyles = makeStyles((theme: Theme) =>
       padding: theme.spacing(0, 1),
       ...theme.mixins.toolbar,
       justifyContent: 'flex-start'
+    },
+    alignRight: {
+      marginLeft: 'auto'
     }
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/ToolPanel.tsx]: x

File: ui/app/src/modules/Preview/Tools/ToolPanel.tsx [hunk 13]
```diff
@@ -73,6 +78,7 @@ export function ToolPanel(props: ToolPanelProps): ReactElement | null {
   const dispatch = useDispatch();
   const { formatMessage } = useIntl();
   const { title, BackIcon, onBack = () => dispatch(popToolsPanelPage()), classes } = props;
+
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/ToolPanel.tsx]: x

File: ui/app/src/modules/Preview/Tools/ToolPanel.tsx [hunk 14]
```diff
@@ -73,6 +78,7 @@ export function ToolPanel(props: ToolPanelProps): ReactElement | null {
...
   return (
     <>
       <PanelHeader
```
Comment (rart) [inline ui/app/src/modules/Preview/Tools/ToolPanel.tsx]: This whole file should disappear from the PR files (check)

File: ui/app/src/components/ToolsPanelPageButton/ToolsPanelPageButton.tsx [hunk 15]
```diff
@@ -25,6 +25,7 @@ export default function ToolsPanelPageButton(props) {
     dispatch(
       pushToolsPanelPage({
         id: 'craftercms.components.ToolsPanelPage',
+        uiKey: `uiKey_${Date.now()}`,
```
Comment (rart) [inline ui/app/src/components/ToolsPanelPageButton/ToolsPanelPageButton.tsx]: 👆🏻 use create widget descriptor method

**Review by rart — APPROVED** [review]

**Pull request merged**